onset_maproom/maproom.py

```diff
@@ -102,33 +106,72 @@ def map_click(click_lat_lng):
     Input("dryDays", "value"),
     Input("drySpell", "value"),
 )
-def onset_plot(click_lat_lng, earlyStartDay, earlyStartMonth, searchDays, wetThreshold, runningDays, runningTotal, minRainyDays, dryDays,drySpell):
+def onset_plots(click_lat_lng, search_start_day, search_start_month, searchDays, wetThreshold,
+                runningDays, runningTotal, minRainyDays, dryDays,drySpell):
     lat, lng = get_coords(click_lat_lng)
-    ds = rr_mrg.sel(X=lng, Y=lat, method="nearest")
-    onsetDays = calc.onset_date(ds.precip, int(earlyStartDay), \
-        calc.strftimeb2int(earlyStartMonth), searchDays, \
-        wetThreshold, runningDays, runningTotal, minRainyDays, dryDays, drySpell)
-    onsetDate = onsetDays["T"] + onsetDays
-    year = pd.DatetimeIndex(onsetDate['T']).year
-    onsetMD = onsetDate.dt.strftime("2000-%m-%d").to_dataframe(name="Onset Date")
-    onsetMD['Year'] = year
-    graph = px.line(
-        data_frame=onsetMD,
-        x="Year", 
-        y="Onset Date",
-    )
-    graph.update_traces(
-        mode="markers+lines",
-        hovertemplate='%{y} %{x}'
-    )
-    graph.update_layout(
-        yaxis=dict(tickformat="%b %d"), 
-        xaxis_title="Year", 
-        yaxis_title="Onset Date",
-        title= f"Starting dates of {int(earlyStartDay)} {earlyStartMonth} season {year.min()}-{year.max()} ({round_latLng(ds.Y)}E, {round_latLng(ds.X)}N)"
-    )
-    return graph
-
+    try:
+        precip = rr_mrg.precip.sel(X=lng, Y=lat, method="nearest", tolerance=0.04)
+    except KeyError:
+        errorFig = pgo.Figure().add_annotation(x=2, y=2,text="No Data to Display",font=dict(family="sans serif",size=30,color="crimson"),showarrow=False, yshift=10, xshift=60)
+        alert1 = dbc.Alert("The point you have chosen is not within the bounding box of this dataset. Please choose a different point.", color="danger", dismissable=True)
+        return errorFig, errorFig, alert1
+    else:
+        precip.load()
+    try:
+        onset_delta = calc.seasonal_onset_date(precip, int(search_start_day),
+            calc.strftimeb2int(search_start_month), int(searchDays),
+            int(wetThreshold), int(runningDays), int(runningTotal),
+            int(minRainyDays), int(dryDays), int(drySpell), time_coord="T")
+    except TypeError:
+        errorFig = pgo.Figure().add_annotation(x=2, y=2,text="No Data to Display",font=dict(family="sans serif",size=30,color="crimson"),showarrow=False, yshift=10, xshift=60)
+        alert1 = dbc.Alert("Please ensure all input boxes are filled for the calculation to run.", color="danger", dismissable=True)
```

Comment:
Check out the `debounce` option. That might help prevent unnecessary form submissions.
https://dash.plotly.com/dash-core-components/input

Reply:
Setting `max` might also help prevent some invalid values, like March 392 as a date.

Reply:
Feel free to propose changes to Kevin's code where appropriate.

In the long run, it would be nice to have a Date input type, with date-specific validation and maybe a convenient picker interface. Dash has a date picker class, but it requires you to pick a year as well as month and day. At some point we should probably extend that to support leaving the year out. Let's not focus on that for now, though. Let's pick the low-hanging fruit (if any) on the front end, and make sure that when back-end validation fails, the user gets good feedback and it's not hugely annoying.